test/json_test.go

```diff
@@ -0,0 +1,83 @@
+package test
+
+import (
+	"github.com/brandonromano/wrecker/test/models"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestPostJSON(t *testing.T) {
+
+	response := models.Response{
+		Content: new(models.User),
+	}
+
+	userIn := models.User{
+		Id:       98,
+		UserName: "Steve Rogers",
+		Location: "New York, NY",
+	}
+
+	_, err := wreckerClient.Post("/users").
+		WithBody(userIn).
+		Into(&response).
+		Execute()
+
+	assert.True(t, err == nil)
+	assert.True(t, response.Success)
+
+	userOut, ok := response.Content.(*models.User)
+
+	assert.True(t, ok)
+	assert.True(t, userOut.Id == 98)
+	assert.True(t, userOut.UserName == "Steve Rogers")
+	assert.True(t, userOut.Location == "New York, NY")
+}
+
+func TestPutJSON(t *testing.T) {
+
+	response := models.Response{
+		Content: new(models.User),
+	}
+
+	userIn := models.User{
+		Id:       99,
+		UserName: "Natasha Romanov",
+		Location: "New York, NY",
+	}
+
+	_, err := wreckerClient.Put("/users").
+		WithBody(userIn).
+		Into(&response).
+		Execute()
+
+	assert.True(t, err == nil)
+	assert.True(t, response.Success)
+
+	userOut, ok := response.Content.(*models.User)
+
+	assert.True(t, ok)
+	assert.True(t, userOut.Id == 99)
+	assert.True(t, userOut.UserName == "Natasha Romanov")
+	assert.True(t, userOut.Location == "New York, NY")
+}
+
+func TestPutString(t *testing.T) {
+
+	response := models.Response{
+		Content: new(string),
+	}
+
+	_, err := wreckerClient.Put("/status").
+		WithBody("status code green").
+		Into(&response).
+		Execute()
+
+	assert.True(t, err == nil)
+	assert.True(t, response.Success)
+
+	status, ok := response.Content.(*string)
+
+	assert.True(t, ok)
+	assert.True(t, *status == "status code green")
```

Comment:
The idea is to JSON-encode everything.  So, if you pass in an object, it gets encoded into a struct.  If you pass in a string, its simply surrounded with double quotes and escaped -- so JSON encoded as a string data.  

The server would still need to handle the response as JSON, but I think that's OK because we're sending the Content-Type as application/json.  In the future, it would be cool to enable other encoding types, like application/xml or even plain text.  But, those would be different encoders, and we'd need to build in a flag to switch between them.  

Does this answer make sense?  And, does this plan work for you?